flocker/common/_filepath.py

```diff
@@ -38,3 +42,55 @@ def make_directory(path):
     if not path.isdir():
         path.makedirs()
     return path
+
+
+class IFilePathExtended(IFilePath):
+    """
+    ``IFilePath`` plus some other ``FilePath`` methods that we need in Flocker.
+    """
+    def remove():
+        """
+        Recursively remove the path and all its descendants.
+        """
+
+    def descendant():
+        """
+        Retrieve a descendant FilePath.
+        """
+
+
+class _TemporaryPath(proxyForInterface(IFilePathExtended, "_path")):
+    """
+    An ``IFilePath`` which when used as a context manager will remove itself.
+    """
+    def __init__(self, path):
+        self._path = path
+
+    @property
+    def path(self):
+        return self._path.path
+
+    def __eq__(self, other):
```

Comment:
I don't think so....but could / should probably add a few more tests for this.
I'm delegating to the wrapped ``FilePath.__eq__`` method which will compare its `.path` byte string  (b"/foo/bar/baz") with `other.path`.